evennia/contrib/rpg/rpsystem/tests.py

```diff
@@ -156,12 +126,8 @@ def test_recog_handler(self):
         self.speaker.recog.add(self.receiver2, recog02)
         self.assertEqual(self.speaker.recog.get(self.receiver1), recog01)
         self.assertEqual(self.speaker.recog.get(self.receiver2), recog02)
-        self.assertEqual(
-            self.speaker.recog.get_regex_tuple(self.receiver1)[0].pattern,
-            "/[0-9]*-*Mr\\ Receiver(?=\\W|$)+|/[0-9]*-*Receiver(?=\\W|$)+|/[0-9]*-*Mr(?=\\W|$)+",
-        )
         self.speaker.recog.remove(self.receiver1)
-        self.assertEqual(self.speaker.recog.get(self.receiver1), sdesc1)
+        self.assertEqual(self.speaker.recog.get(self.receiver1), None)
```

Comment:
This is a backwards-incompatible change, for what seems to be rather limited gain. Is there a reason why we can't just return the sdesc/key if no recog was registered?

Reply:
I admit I hadn't considered backwards compatability, but yes. Using the sdesc/key as a fallback for recog makes the separation of `process_sdesc` and `process_recog` completely meaningless, as the recog handler takes precedence to the sdesc handler. As a result, everything would processed via `process_recog` if the viewing object has a recog handler, regardless of whether it was a recog or an sdesc, and `process_sdesc` would only be used by viewing objects that have no recog handler.

The issue isn't noticeable in the original code because `process_recog` just calls `process_desc`, making them indistinguishable, but caused me a lot of headaches when I tried to add a separate format for recogs versus unrecognized sdescs. Returning `None` if there's no recog and explicitly coding in a fallback in the new `get_sdesc` hook are how I ultimately resolved that.

Reply:
Hm. Yes, I can see the argument of giving the implementer more power to separate between recog/non-recog'ed desciptors for games that want to do that. Thanks for elaborating! 
It's not ideal from a compatibility perspective though, so this will need to be properly documented in the CHANGELOG (I can do that when this is to be merged, or you can do it if you can sum changes into max two lines).